include/circt/Dialect/Moore/MooreTypes.td

```diff
@@ -52,6 +52,34 @@ def EventType : MooreTypeDef<"Event", [], "moore::UnpackedType"> {
   let summary = "the SystemVerilog `event` type";
 }
 
+//===----------------------------------------------------------------------===//
+// ClassHandleType
+//===----------------------------------------------------------------------===//
+
+// !moore.class.object<@C>
+// Encodes a handle of class instance living on the program's heap,
+// whose dynamic layout begins with a vptr.
+// The parameter is the class symbol (e.g., @C).
+
+def ClassHandleType : MooreTypeDef<"ClassHandle", [], "moore::UnpackedType"> {
+  let mnemonic = "class.object";
+  let summary = "Class handle type";
+  let description = [{
+    Class handle type representing `!moore.class.object<@C>`. The parameter is a
+    SymbolRef to the class declaration symbol. The handle is opaque here.
+  }];
+  let parameters = (ins AttrParameter<"::mlir::SymbolRefAttr", "class symbol">:$classSym);
+  let assemblyFormat = "`<` $classSym `>`";
+  let extraClassDeclaration = [{
+    /// True if this handle’s class is the same as, or derived from, `baseName`.
+    /// `baseName` must be the root symbol name (e.g. @"pkg::Class").
+    bool isSameOrDerivedFrom(::mlir::Operation *anchor,
+                             ::mlir::SymbolRefAttr baseName);
+    /// Find the lowest ancestor of this class which has a property `fieldName`.
+    ClassHandleType ancestorWithProperty(Operation *anchor,
+                                         StringRef fieldName);
+  }];
```

Comment:
Are these ever used? It's usually a good idea to only need the functionality that is immediately needed and useful for the PR.

Reply:
Fair enough; they are a remnant from splitting commits 😄